src/charm.py

```diff
@@ -367,6 +368,16 @@ def _on_config_changed(self, event: EventBase):
         """Handler for config changed events."""
         if not self.unit.is_leader():
             return
+        if self._hugepages_is_enabled():
+            if not self._cpu_is_compatible_for_hugepages():
+                raise IncompatibleCPUError(
+                    "\nCPU is not compatible!\n"
+                    "Please use a CPU that has the following capabilities: "
+                    f"{', '.join(REQUIRED_CPU_EXTENSIONS + REQUIRED_CPU_EXTENSIONS_HUGEPAGES)}"
+                )
+            if not self._hugepages_are_available():
+                self.unit.status = BlockedStatus("Not enough HugePages available")
```

Comment:
I think we should also observe `update_status` because no event may be triggered when finally hugepages are available

Reply:
This method checks that at least 2Gi of HugePages are allocatable. For this value to change, the user should either restart the host or the k8s node, causing the pod to also restart.

Reply:
@gruyaume I can add `event.defer()` here, as in the worst case the event will be re-run when `update-status` is fired